### ISSUE TITLE
SS 4 - Parse error: syntax error, unexpected ':', expecting ';' or '{' in /public_html/vendor/symfony/filesystem/Filesystem.php on line 727

### ISSUE THREAD

**Issue by SqualaDesign** [silverstripe/silverstripe-framework#7709]
## Affected Version
SilverStripe 4.0.1

## Description
I'm trying to install SilverStripe 4.0.1 on my VPS, LAMP technology. All requirements are correct.
![screencapture-205-134-253-41-devsquala-install-php-1513307543231](https://user-images.githubusercontent.com/8541984/34025106-eef0c3de-e1b2-11e7-9624-460680aac7da.png)

## Steps to Reproduce
I'm using the normal installation and not composer.

# PR:

- [ ] https://github.com/silverstripe/cow/pull/66

**Comment by tractorcow** [comment]
This is a know bug and will be fixed in 4.0.2. The issue is purely with the zip generation, which was optimised for php 7.1.

If you are running php 7.0 or 5.6 you will run into errors where 7.1 only API is used.

You can fix this by clearing your vendor folder and re-installing the dependencies with composer. This will optimise the installed libraries for your local platform php version.

**Comment by tractorcow** [comment]
Relevant cow issue: https://github.com/silverstripe/cow/issues/65

**Comment by SqualaDesign** [comment]
I'm running PHP 5.6 and I never use composer before :(
When the vers. 4.0.2 will be available?
Thanks

**Comment by tractorcow** [comment]
Early next year;

In the short term I would suggest you setup composer https://getcomposer.org/ and try it out. :) I promise it will make development easier for you.  :)

**Comment by SqualaDesign** [comment]
I will definitely try, thanks

**Comment by dhensby** [comment]
My understanding is that composer is not meant to be optional for SS 4 :P

**Comment by tractorcow** [comment]
I think we should make sure the zip archive works in as many platforms as possible.

of course, we could end up with the inverse issue, where we ship a php 5.6 version of a dependency that doesn't work on php 7.1 / 7.2.

**Comment by BloodRedOwl** [comment]
I'm running php 7.1 and installed using composer and still getting this issue.  Would anyone know a solution to it?

**Comment by dhensby** [comment]
> I would totally rather drop zip download for 5.x onwards

Can we not do that now? As I said, my understanding is composer is now a requirement for SilverStripe...

> I would totally rather drop zip download for 5.x onwards, maybe. We could replace it with an installer that sets up composer + creates your project for you, maybe?

This stuff is the responsibility of the devs.

**Comment by tractorcow** [comment]
> Can we not do that now? As I said, my understanding is composer is now a requirement for SilverStripe...

Yes, and, if we are using composer, we don't need the zip downloads. I don't understand your point sorry.

**Comment by dhensby** [comment]
Let's not wait for SS5 to stop the zips :)

**Comment by tractorcow** [comment]
Oh I see, that's better then!

**Comment by tractorcow** [comment]
I can are english best!

**Comment by tractorcow** [comment]
Fix at https://github.com/silverstripe/cow/pull/66

**Comment by kigi001** [comment]
I used composer for the first time today and it installed the SS4.1.0 version and chose php7.19 (WAMP) - by the way, where can I actually find the version I installed? (I know it by watching the installation, since SS4 I can't find the silverstripe_version file anymore).
Anyway, I get the following message in cmd:
 unexpected ':', expecting ';' or '{' in D:\kd\websites\dev\composer\vendor\symfony\filesystem\Filesystem.php on line 732

**Comment by kigi001** [comment]
Ups, I just noticed that my WAMP actually wasn't set to php7.19 itself. I changed this and re-installed the project. Now it went through.
I pressed enter in the cmd and now it added the following:
unexpected ':', expecting ';' or '{' in D:\kd\websites\dev\composer\vendor\symfony\filesystem\Filesystem.php on line 732
'unexpected' is not recognized as an internal or external com
ged this and re-installed the project. Now it went through.
I pressed enter in the cmd and now it added the following:
unexpected ':', expecting ';' or '{' in dev\composer\vendor\symfony\filesystem\Filesystem.php on line 732
'unexpected' is not recognized as an internal or external command,
operable program or batch file.
